test/web-platform-tests/to-run.yaml

```diff
@@ -824,3 +824,15 @@ failure.html: [fail, Fetch is not implemented]
 toascii.window.html: [fail, Unknown]
 url-constructor.html: [fail, url.search vs. url.searchParams propagation]
 urlencoded-parser.html: [fail, Fetch is not implemented]
+
+---
+
+DIR: websockets
+
+Create-Secure-extensions-empty.htm: [fail, Buggy test as the test does not take into account the mandatory permessage-deflate extension]
+Secure-Close-1005-verify-code.htm: [fail, See https://github.com/websockets/ws/issues/1257]
+closing-handshake/003.html*: [fail, See https://github.com/websockets/ws/issues/1257]
+interfaces/WebSocket/close/close-basic.html*: [fail, Mutates globals]
+interfaces/WebSocket/close/close-connecting.html*: [fail, Potentially buggy test as Chrome fails it too]
```

Comment:
Firefox doesn’t fail it locally, however (even though [wpt.fyi](https://wpt.fyi/websockets/interfaces/WebSocket/close/close-connecting.html) says otherwise). Fixing it seems to require digging through ws module's guts, which I didn't do.

Reply:
This [assertion](https://github.com/w3c/web-platform-tests/blob/079f031924c84818c689349d0c45fbf3c7833ba8/websockets/interfaces/WebSocket/close/close-connecting.html#L16) fails in ws if `close()` is called when connecting as the close `'event'` is emitted on the same tick.

Reply:
The patch does not fix the issue, unfortunately :( Seems like the actual error thrown is

```
Error: Parse Error
    at Socket.socketOnData (_http_client.js:439:20)
    at Socket.emit (events.js:159:13)
    at addChunk (_stream_readable.js:265:12)
    at readableAddChunk (_stream_readable.js:252:11)
    at Socket.Readable.push (_stream_readable.js:209:10)
    at TCP.onread (net.js:608:20)
```

I assume we can't do much about this.

Reply:
Hmm this is weird.